extensions/wikia/MercuryApi/MercuryApiHooks.class.php

```diff
@@ -144,4 +146,51 @@ static public function onCuratedContentSave( $sections ) {
 	private static function encodeURIQueryParam( $str ) {
 		return strtr( rawurlencode( $str ), [ '%21' => '!', '%27' => "'", '%28' => '(', '%29' => ')', '%2A' => '*' ] );
 	}
+
+	/**
+	 * @desc Adds <section class="mobile-hidden"></section> wrapper for sections
+	 * @param $skin
+	 * @param $level
+	 * @param $attribs
+	 * @param $anchor
+	 * @param $text
+	 * @param $link
+	 * @param $legacyAnchor
+	 * @param $ret
+	 */
+	public static function onMakeHeadline( $skin, $level, $attribs, $anchor, $text, $link, $legacyAnchor, &$ret ){
+		global $wgArticleAsJson;
+
+		// this is pseudo-versioning query param for collapsible sections (XW-4393)
+		// should be removed after all App caches are invalidated
+		if ( !empty( RequestContext::getMain()->getRequest()->getVal( 'collapsibleSections' ) ) &&
+		     $wgArticleAsJson && $level == 2
```

Comment:
does `wgArticleAsJson` ensure that this section tag is added only in MercuryApi I'm not sure but mobileApps may use this as well